engine/src/main/java/org/camunda/bpm/engine/impl/interceptor/CommandContextInterceptor.java

```diff
@@ -88,9 +88,9 @@ public <T> T execute(Command<T> command) {
 
     // only create a new command context on the current command level (CAM-10002)
     boolean isNew = ProcessEngineContextImpl.consume();
-    boolean openNew = (context == null || isNew);
+    boolean openNew = (context == null || isNew); // will be true only for outer commands
```

Comment:
> will be true only for outer commands

❓Why?

Reply:
This line holds true only for outer commands and i've added a comment to make the life of the reader easier. I've verified that with debugging.

Reply:
Daniel might have a point here 🤔 

When the context is `null`, I agree that this indicates a call from a top-level command since child commands access the already existing command context of the top-level command. However, `isNew` can be controlled by the user and is not an indicator for a top-level command. We need to change that. ❌